glGetStringi

juj opened the pull request:
Implement glGetStringi() for GLES3 compatibility and add a test. Closes #3472.

kripken commented on src/library_gl.js:
```diff
@@ -963,6 +976,50 @@ var LibraryGL = {
+      GL.recordError(0x0502 /* GL_INVALID_OPERATION */); // Calling GLES3/WebGL2 function with a GLES2/WebGL1 context
+      return 0;
+    }
+    if (GL.stringiCache[name]) {
```
maybe store `GL.stringiCache[name]` to a local? it's used 3 times.

kripken commented:
This is our first WebGL2 test - it will fail on some of the bots, I think?

lgtm aside from this and the one comment above.

juj commented:
Addressed the review. The test is soft in a sense that if WebGL 2 is not supported, then the test is just ignored and passed directly: https://github.com/kripken/emscripten/pull/3725/files#diff-d86cc47838f8ae6a41f09e704c1cbd9aR21
